### PR TITLE
Update Even Better Mod Descriptions

### PR DESCRIPTION
**Pull request by IllusiveMan196**
Updating EBMD data to reflect both the new main version and my fixes. Yes, Wrye Bash patcher errored out again.

### CODE REVIEW

File: masterlist.yaml [hunk 1]
```diff
@@ -2784,31 +2784,102 @@ plugins:
...
-        condition: 'checksum("EBMD_AllInOne.esp", C0BEAF59)'
+        condition: 'checksum("EBMD_AllInOne.esp", F0DDE373) or checksum("EBMD_AllInOne.esp", 08B0A737)'
     clean:
-      - crc: 0x504E4A32
+      # Original
+      - crc: 0x50AC05D1
+        util: 'FO4Edit v4.0.4'
+      # Fixed
+      - crc: 0xCFD5B0B2
         util: 'FO4Edit v4.0.4'
```
Comment (MacSplody) [inline masterlist.yaml:2873]: Keep newline between mods.
```suggestion
        util: 'FO4Edit v4.0.4'

```

File: masterlist.yaml [hunk 2]
```diff
@@ -2784,31 +2784,102 @@ plugins:
         name: 'Even Better Mod Descriptions'
     msg:
```
Comment (MacSplody) [inline masterlist.yaml]: You could add a link to your mod in locations.

```suggestion
        name: 'Even Better Mod Descriptions'
      - link: 'https://www.nexusmods.com/fallout4/mods/74459/'
        name: 'Illusive Fixes and Merges'
    msg:
```

File: masterlist.yaml [hunk 3]
```diff
@@ -2784,31 +2784,102 @@ plugins:
...
+      # GOTY (No Perks 4.1) Fixed
+      - crc: 0xBDD21B9E
+        util: 'FO4Edit v4.0.4'
+      # GOTY (No Perks 4.1) Fixed UFO4P
+      - crc: 0x08B0A737
+        util: 'FO4Edit v4.0.4'
+  - name: 'EvenBetterMDesc.esp'
+    url:
+      - link: 'https://www.nexusmods.com/fallout4/mods/36105/'
+        name: 'Even Better Mod Descriptions'
```
Comment (MacSplody) [inline masterlist.yaml:2811]: You could also combine all the location data into one regex at the top of the mod files.
```yaml
  - name: '(EBMD(_AllInOne|-UF04P Patch)|Even(BetterMDesc|BMD_(Automatron|FarHarbor|NukaWorld)DLC))\.esp'
    url:
      - link: 'https://www.nexusmods.com/fallout4/mods/36105/'
        name: 'Even Better Mod Descriptions'
      - link: 'https://www.nexusmods.com/fallout4/mods/74459/'
        name: 'Illusive Fixes and Merges
```

File: masterlist.yaml [hunk 4]
```diff
@@ -2784,31 +2784,102 @@ plugins:
...
+    msg:
+      - <<: *alsoUseX
+        subs: [ '[Even Better Mod Descriptions Fixed](https://www.nexusmods.com/fallout4/mods/74459/)' ]
+        condition: 'checksum("EvenBetterMDesc.esp", 31185F01)'
```
Comment (MacSplody) [inline masterlist.yaml]: If you add version info to your files, you could replace the `checksum()` with `version()`.
It's far easier to manage than checksums and is easy enough to add version info through xEdit.

```yaml
        condition: 'checksum("EvenBetterMDesc.esp", 31185F01)'
```

Comment (MacSplody) [reply masterlist.yaml]: I can show you on discord if you're interested.

File: masterlist.yaml (hunk 3 above)
Comment (IllusiveMan196) [inline masterlist.yaml]: What to do with warnings then? They're mostly dependent on what file is installed

**Comment by IllusiveMan196** [comment]
Regex isn't acceptable input for plugin name check?

File: masterlist.yaml [hunk 5]
```diff
@@ -2778,35 +2778,72 @@ plugins:
       - Delev
       - Relev
 
-  - name: 'EBMD_AllInOne.esp'
+  - name: '(EBMD(_AllInOne|-UF04P Patch)|Even(BetterMDesc|BMD_(Automatron|FarHarbor|NukaWorld)DLC))\.esp'
     url:
       - link: 'https://www.nexusmods.com/fallout4/mods/36105/'
         name: 'Even Better Mod Descriptions'
+      - link: 'https://www.nexusmods.com/fallout4/mods/74459/'
+        name: 'Illusive Fixes and Merges'
```
Comment (MacSplody) [inline masterlist.yaml]:
```suggestion
  - name: '(EBMD(_AllInOne|-UF04P Patch)|Even(BetterMDesc|BMD_(Automatron|FarHarbor|NukaWorld)DLC))\.esp'
    url:
      - link: 'https://www.nexusmods.com/fallout4/mods/36105/'
        name: 'Even Better Mod Descriptions'
      - link: 'https://www.nexusmods.com/fallout4/mods/74459/'
        name: 'Illusive Fixes and Merges'
  - name: 'EBMD_AllInOne.esp'
```

**Comment by IllusiveMan196** [comment]
alright... when will it work?

File: masterlist.yaml [hunk 6]
```diff
@@ -2778,35 +2778,110 @@ plugins:
...
+    url:
+      - link: 'https://www.nexusmods.com/fallout4/mods/36105/'
+        name: 'Even Better Mod Descriptions'
+      - link: 'https://www.nexusmods.com/fallout4/mods/74459/'
+        name: 'Illusive Fixes and Merges'
   - name: 'EBMD_AllInOne.esp'
+    msg:
+      - <<: *alsoUseX
+        subs: [ '[Even Better Mod Descriptions Fixed](https://www.nexusmods.com/fallout4/mods/74459/)' ]
+        condition: 'not version("EBMD_AllInOne.esp", "4.1", =)'
```
Comment (pStyl3) [inline masterlist.yaml]: The available comparators are: [Link](https://loot-api.readthedocs.io/en/stable/metadata/conditions.html#types)
A single `=` is not a valid comparator.

**Review by pStyl3 — DISMISSED** [review]

File: masterlist.yaml [hunk 7]
```diff
@@ -2778,35 +2778,110 @@ plugins:
...
-    # Can skip adding other as this is the main variant and has most downloads.
   - name: 'EBMD-UF04P Patch.esp'
     url:
       - link: 'https://www.nexusmods.com/fallout4/mods/36105/'
         name: 'Even Better Mod Descriptions'
     msg:
       - <<: *alreadyInOrFixedByX
         subs: [ 'Even Better Mod Descriptions GOTY Fixed](https://www.nexusmods.com/fallout4/mods/74459/)' ]
-        condition: 'checksum("EBMD_AllInOne.esp", C0BEAF59)'
+        condition: 'version("EBMD_AllInOne.esp", "4.1", =, >=)'
```
Comment (MacSplody) [inline masterlist.yaml]:
```suggestion
        condition: 'version("EBMD_AllInOne.esp", "4.1", >=)'
```

File: masterlist.yaml [hunk 8]
```diff
@@ -2778,35 +2778,110 @@ plugins:
...
         util: 'FO4Edit v4.0.4'
       # Fixed
-      - crc: 0xC0BEAF59
+      - crc: 0xBEB20B31
         util: 'FO4Edit v4.0.4'
-    # Can skip adding other as this is the main variant and has most downloads.
   - name: 'EBMD-UF04P Patch.esp'
     url:
       - link: 'https://www.nexusmods.com/fallout4/mods/36105/'
         name: 'Even Better Mod Descriptions'
```
Comment (MacSplody) [inline masterlist.yaml]:
```suggestion
  - name: 'EBMD-UF04P Patch.esp'
```

File: masterlist.yaml [hunk 9]
```diff
@@ -2778,35 +2778,110 @@ plugins:
...
+      # Original
+      - crc: 0x38F38215
+        util: 'FO4Edit v4.0.4'
+      # Fixed
+      - crc: 0xA1D256BB
+        util: 'FO4Edit v4.0.4'
+  - name: 'EvenBMD_NukaWorldDLC.esp'
+    url:
+      - link: 'https://www.nexusmods.com/fallout4/mods/36105/'
+        name: 'Even Better Mod Descriptions'
```
Comment (MacSplody) [inline masterlist.yaml]:
```suggestion
  - name: 'EvenBMD_NukaWorldDLC.esp'
```

File: masterlist.yaml [hunk 10]
```diff
@@ -2778,35 +2778,110 @@ plugins:
...
         util: 'FO4Edit v4.0.4'
-      # Merged with UFO4P patch as suggested on the mod page
-      - crc: 0xDC398CCF
+      # Fixed
+      - crc: 0x384AF106
+        util: 'FO4Edit v4.0.4'
+  - name: 'EvenBMD_FarHarborDLC.esp'
+    url:
+      - link: 'https://www.nexusmods.com/fallout4/mods/36105/'
+        name: 'Even Better Mod Descriptions'
```
Comment (MacSplody) [inline masterlist.yaml]:
```suggestion
  - name: 'EvenBMD_FarHarborDLC.esp'
```

**Review by MacSplody — CHANGES_REQUESTED** [review]
One condition to fix, and you can remove the extra location data.

**Comment by IllusiveMan196** [comment]
Done

File: masterlist.yaml [hunk 11]
```diff
@@ -2778,35 +2778,98 @@ plugins:
...
-    # Can skip adding other as this is the main variant and has most downloads.
   - name: 'EBMD-UF04P Patch.esp'
-    url:
-      - link: 'https://www.nexusmods.com/fallout4/mods/36105/'
-        name: 'Even Better Mod Descriptions'
     msg:
       - <<: *alreadyInOrFixedByX
         subs: [ 'Even Better Mod Descriptions GOTY Fixed](https://www.nexusmods.com/fallout4/mods/74459/)' ]
-        condition: 'checksum("EBMD_AllInOne.esp", C0BEAF59)'
+        condition: 'version("EBMD_AllInOne.esp", "4.1", >=)'
```
Comment (MacSplody) [inline masterlist.yaml:2866]: Whoops that should have a `not`, and the filename is wrong.

```suggestion
        condition: 'not version("EBMD-UF04P Patch.esp", "4.1", >=)'
```

File: masterlist.yaml [hunk 12]
```diff
@@ -2778,35 +2778,98 @@ plugins:
...
-    # Can skip adding other as this is the main variant and has most downloads.
   - name: 'EBMD-UF04P Patch.esp'
-    url:
-      - link: 'https://www.nexusmods.com/fallout4/mods/36105/'
-        name: 'Even Better Mod Descriptions'
     msg:
       - <<: *alreadyInOrFixedByX
         subs: [ 'Even Better Mod Descriptions GOTY Fixed](https://www.nexusmods.com/fallout4/mods/74459/)' ]
-        condition: 'checksum("EBMD_AllInOne.esp", C0BEAF59)'
+        condition: 'not version("EBMD-UF04P Patch.esp", "4.1", >=)'
```
Comment (MacSplody) [inline masterlist.yaml]: Sigh too sleepy.
```suggestion
        condition: 'version("EBMD_AllInOne.esp", "4.1", >=)'
```

**Review by MacSplody — APPROVED** [review]

**Comment by MacSplody** [comment]
Thanks.

**Pull request merged**